### PR TITLE
add vtbackup reference and refactor backup/restore docs

### PR DESCRIPTION
**Pull request by maxenglander**
Accompanies vitessio/vitess#11979.

In order to document all of the new metrics introduced by vitessio/vitess#11979, I would like to be able to talk about `vtbackup`. Currently the docs don't mention `vtbackup` at all. This PR reorganizes adds a reference page for vtbackup and re-organizes the backup/restore docs a bit to make it easier to talk about `vtbackup`.

### CODE REVIEW

File: content/en/docs/16.0/reference/programs/vtbackup.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
+---
```
Comment (rsajwani) [inline content/en/docs/16.0/reference/programs/vtbackup.md:1]: I actually have another PR for vtbackup .. but I will go through yours and if needed will close mine ..

Comment (maxenglander) [reply content/en/docs/16.0/reference/programs/vtbackup.md:1]: If you have another one feel free to go with that and close this one! I have no preference, whichever you prefer.

File: content/en/docs/16.0/reference/programs/vtbackup.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+ - Provisioning as much disk space for `vtbackup` as would be given to vttablet.
+   The data directory MUST be empty at startup. Do NOT reuse a persistent disk.
+ - Running `vtbackup` periodically for each shard, for each backup storage location.
+ - Ensuring that at most one instance runs at a time for a given pair of shard
+   and backup storage location.
+ - Retrying `vtbackup` if it fails.
+ - Alerting human operators if the failure is persistent.
+
+## Example Usage
+
```
Comment (rsajwani) [inline content/en/docs/16.0/reference/programs/vtbackup.md:24]: minor: Consider you have running Vitess cluster, the following command will create backup of keyspace `commerce` having shard `0`

**Review by rsajwani — APPROVED** [review]
Overall looks good.. There are few deprecation in the flags but I have taken care of it in another PR (https://github.com/vitessio/website/pull/1330/files)... once you push this change .. I will apply my PR on top of it.

File: content/en/docs/16.0/reference/programs/vtbackup.md (hunk 2 above)
Comment (rsajwani) [reply content/en/docs/16.0/reference/programs/vtbackup.md:24]: I have taken care of this in my PR.

**Pull request merged**